roles/ansible-deploy-rudder/library/rudder_node_settings.py

```diff
@@ -0,0 +1,541 @@
+#     format='[%(asctime)s %(levelname)s %(module)s - %(funcName)s: %(message)s',
+#     datefmt='%Y-%m-%d %H:%M:%S]',
+#     encoding='utf-8',
+#     level=logging.DEBUG
+# )
+# }}}
+
+# Disable SSL certificat warning messages
+requests.packages.urllib3.disable_warnings(
+    InsecureRequestWarning
+    )
+
+DOCUMENTATION = '''
+---
+module: rudder_node_settings
+author:
+  - Normation
+version_added: '2.9'
+short_description: Configure Rudder Nodes 6.2 parameters via APIs
+requirements:
+    - 'python >= 2.7'
+
+options:
+
+  rudder_url:
+    description:
+      - Providing Rudder server IP address. Defaults to localhost of the target node if not set, with certificate validation disabled, unless explicitly enabled by setting validate_certs.
+    required: false
+    type: str
+
+  rudder_token:
+    description:
+      - Providing Rudder server token. Defaults to the content of /var/rudder/run/api-token if not set.
+    required: false
+    type: str
+
+  validate_certs:
+    description:
+      - Choosing either to ignore or not Rudder certificate validation. Defaults to true.
+    required: false
+    type: boolean
+    
+  node_id:
+    description:
+      - Define the identifier of the node to be configured
+    required: true
+    type: str
+
+  policy_mode:
+    description:
+      - Set the policy mode to (default, enforce or audit)
+    required: false
+    type: str
+    
+  state:
+    description:
+      - Set the node life cycle state to (enabled, ignored, empty-policies, initializing or preparing-eol)
+    required: false
+    type: str
+    
+  properties:
+    description:
+      - Define a properties (with "name:" and "value:")
+    required: false
+    type: dict
+    
+  agent_key:
+    description:
+      - Define information about agent key or certificate (update PEM with ("value:<PEM>") and status with "status:<certified|undefined>")
+    required: false
+    type: dict
+    
+'''
+
+EXAMPLES = r'''
+  - name: Simple Modify Rudder Node Settings
+    rudder_node_settings:
+        rudder_url: "https://my.rudder.server/rudder"
+        node_id: my_node_id
+        policy_mode: enforce
+        
+  - name: Complex Modify Rudder Node Settings
+    rudder_node_settings:
+        rudder_url: "https://my.rudder.server/rudder"
+        rudder_token: "<rudder_server_token>"
+        node_id: root
+        pending: accepted
+        policy_mode: audit
+        state: enabled
+        properties:
+          name: "env_type"
+          value: "production"
+        validate_certs: False
+'''
+
+class RudderNodeSettingsInterface(object):
+    def __init__(self, module):
+        self._module = module
+        self.headers = {
+            "Content-Type": "application/json"
+        }
+        self.validate_certs = True
+        # Get local API Token (when is not specified)
+        if module.params.get('rudder_token', None):
+            self.headers = {"X-API-Token": module.params['rudder_token']}
```

Comment:
you are missing an application/json here

Reply:
This is now fixed, thanks for the feedback, it should fix the problem of sending the data.